functions/time/gmtime.c

```diff
@@ -1,16 +1,111 @@
 /* gmtime( const time_t * )
-
-   This file is part of the Public Domain C Library (PDCLib).
-   Permission is granted to use, modify, and / or redistribute at will.
+ * From newlib gmtime_r.c
+ * Original Author: Adapted from tzcode maintained by Arthur David Olson.
+ *
+ *  This file is part of the Public Domain C Library (PDCLib).
+ *  Permission is granted to use, modify, and / or redistribute at will.
 */
 
 #include <time.h>
 
 #ifndef REGTEST
+/* Move epoch from 01.01.1970 to 01.03.0000 (yes, Year 0) - this is the first
+ * day of a 400-year long "era", right after additional day of leap year.
+ * This adjustment is required only for date calculation, so instead of
+ * modifying time_t value (which would require 64-bit operations to work
+ * correctly) it's enough to adjust the calculated number of days since epoch.
+ */
+#define EPOCH_ADJUSTMENT_DAYS	719468L
+/* year to which the adjustment was made */
+#define ADJUSTED_EPOCH_YEAR	0
+/* 1st March of year 0 is Wednesday */
+#define ADJUSTED_EPOCH_WDAY	3
+/* there are 97 leap years in 400-year periods. ((400 - 97) * 365 + 97 * 366) */
+#define DAYS_PER_ERA		146097L
+/* there are 24 leap years in 100-year periods. ((100 - 24) * 365 + 24 * 366) */
+#define DAYS_PER_CENTURY	36524L
+/* there is one leap year every 4 years */
+#define DAYS_PER_4_YEARS	(3 * 365 + 366)
+/* number of days in a non-leap year */
+#define DAYS_PER_YEAR		365
+/* number of days in January */
+#define DAYS_IN_JANUARY		31
+/* number of days in non-leap February */
+#define DAYS_IN_FEBRUARY	28
+/* number of years per era */
+#define YEARS_PER_ERA		400
+
+#define SECSPERMIN	60L
+#define MINSPERHOUR	60L
+#define HOURSPERDAY	24L
+#define SECSPERHOUR	(SECSPERMIN * MINSPERHOUR)
+#define SECSPERDAY	(SECSPERHOUR * HOURSPERDAY)
+#define DAYSPERWEEK	7
+#define MONSPERYEAR	12
+
+#define YEAR_BASE	1900
+#define EPOCH_YEAR      1970
+#define EPOCH_WDAY      4
+#define EPOCH_YEARS_SINCE_LEAP 2
+#define EPOCH_YEARS_SINCE_CENTURY 70
+#define EPOCH_YEARS_SINCE_LEAP_CENTURY 370
+
+#define isleap(y) ((((y) % 4) == 0 && ((y) % 100) != 0) || ((y) % 400) == 0)
+
+
+
+static struct tm res;
 
-struct tm * gmtime( const time_t * timer )
+struct tm * gmtime( const time_t * tim_p )
```

Comment:
I assume this breaks consistency with the header. I'd recommend to keep the pdclib variant for consistency with other functions (I also assume it's closer to C spec?).

Reply:
Done